edge/pkg/metamanager/client/serviceaccount.go

```diff
@@ -21,20 +26,103 @@ type ServiceAccountTokenGetter interface {
 // ServiceAccountTokenInterface is interface for client service account token
 type ServiceAccountTokenInterface interface {
 	GetServiceAccountToken(namespace string, name string, tr *authenticationv1.TokenRequest) (*authenticationv1.TokenRequest, error)
+	DeleteServiceAccountToken(podUID types.UID)
 }
 
 type serviceAccountToken struct {
 	send SendInterface
 }
 
+const maxTTL = 24 * time.Hour
+
 func newServiceAccountToken(s SendInterface) *serviceAccountToken {
 	return &serviceAccountToken{
 		send: s,
 	}
 }
 
-func (c *serviceAccountToken) GetServiceAccountToken(namespace string, name string, tr *authenticationv1.TokenRequest) (*authenticationv1.TokenRequest, error) {
-	resource := fmt.Sprintf("%s/%s/%s", namespace, model.ResourceTypeServiceAccountToken, name)
+// Resource format: <namespace>/<restype>/<resid>/<refType>/<refName>/<refUID>
+// return refUID
+func parseServiceAccountRefResource(resource string) types.UID {
+	tokens := strings.Split(resource, constants.ResourceSep)
+	switch len(tokens) {
+	case 6:
+		return types.UID(tokens[len(tokens)-1])
+	default:
+	}
+	return ""
+}
+
+func (c *serviceAccountToken) DeleteServiceAccountToken(podUID types.UID) {
+	svcAccounts, err := dao.QueryAllMeta("type", model.ResourceTypeServiceAccountToken)
```

Comment:
QueryAllMeta performance may be relatively low, could we use sql like to query

Reply:
New issue to trace.